pilot/pkg/model/validation.go

```diff
@@ -824,9 +824,102 @@ func validateLoadBalancer(settings *networking.LoadBalancerSettings) (errs error
 		}
 	}
 
+	srcLocalities := []string{}
+
+	for _, locality := range settings.LocalitySettings.GetDistribute() {
+		srcLocalities = append(srcLocalities, locality.From)
+		var totalWeight uint32
+		destLocalities := []string{}
+		for loc, weight := range locality.To {
+			destLocalities = append(destLocalities, loc)
+			if weight == 0 {
+				errs = appendErrors(errs, fmt.Errorf("locality weight must not be in range [1, 100]"))
+			}
+			totalWeight += weight
+		}
+		if totalWeight != 100 {
+			errs = appendErrors(errs, fmt.Errorf("total locality weight %v != 100", totalWeight))
+		}
+		errs = appendErrors(errs, validateLocalities(destLocalities))
+	}
+
+	errs = appendErrors(errs, validateLocalities(srcLocalities))
+
+	for _, failover := range settings.LocalitySettings.GetFailover() {
+		if failover.From == failover.To {
+			errs = appendErrors(errs, fmt.Errorf("locality lb failover settings must specify different regions"))
+		}
```

Comment:
also add validation to make sure failover.To is only a string, with no wildcards. or it can be just *. But not things like region*